project_timesheet_management/models/timesheet_invoice.py

```diff
@@ -0,0 +1,764 @@
+                    mail_template.write({
+                        'email_to': user_id.email or '',
+                        'email_from': self.env.user.email or '',
+                        'reply_to': user_id.email or ''})
+                    mail_template.sudo().with_context(state=state).send_mail(
+                    invoice.id, force_send=True)
+
+    def button_completed(self):
+        """
+        Set timesheet invoice to completed.
+        """
+        su_po_group = "project_management_security.group_supp_project_coordinator"
+        mail_template = self.env.ref(
+            "project_timesheet_management.email_timesheet_invoice_confirmed")
+        it_manager = "project.group_project_manager"
+        author_id = (
+            self.env["res.users"].sudo().browse(request.session.uid).partner_id.id
+        )
+        for timesheet_inv in self:
+            timesheet_inv.with_context(from_approval=True).sudo().write(
+                {
+                    "state": "completed",
+                    "completed_date": fields.Date.context_today(timesheet_inv),
+                    "completed_by": self.env.user.id,
+                }
+            )
+            state = dict(timesheet_inv.fields_get(
+                    allfields=['state'])['state']['selection'])[timesheet_inv.state]
+            mail_template.write({
+                'email_to': timesheet_inv.project_id.product_owner_id.email or '',
+                'reply_to': timesheet_inv.project_id.product_owner_id.email or '',
+                'email_from': self.env.user.email or self.env.user.partner_id.email,
+                })
+            mail_template.sudo().with_context(state=state).send_mail(
+                    timesheet_inv.id,force_send=True,
+                    email_values={"author_id": author_id or False})
+
+            # supplier project coordinator
+            project_id = timesheet_inv.sudo().project_id
+            user_ids = project_id.favorite_user_ids.filtered(
+                lambda user: user.has_group(su_po_group)
+                             and user.id != timesheet_inv.sudo().project_id.product_owner_id.id
+                             and (
+                                 user.partner_id == timesheet_inv.sudo().partner_id
+                                 or user.supplier_company_id == timesheet_inv.sudo().partner_id
+                             )
+            ) or project_id.resource_ids.mapped("user_ids").filtered(
+                lambda user: user.has_group(su_po_group)
+                             and user.id != timesheet_inv.sudo().project_id.product_owner_id.id
+                             and (
+                                 user.partner_id == timesheet_inv.sudo().partner_id
+                                 or user.supplier_company_id == timesheet_inv.sudo().partner_id
+                             )
+            )
+
+            for user_id in user_ids:
+                mail_template.write({
```

Comment:
You should not write to the email template, instead you can use email_values as parameter for send_mail function and no need to define reply_to  separately as it uses the same value as email_from